app/Cargo.toml

```diff
@@ -22,11 +24,14 @@ attestation_agent = { path = "../", default-features = false }
 async-trait = "0.1.56"
 
 [build-dependencies]
-tonic-build = "0.7.2"
+tonic-build = { version = "0.7.2", optional = true }
+ttrpc-codegen = { version = "0.4.1", optional = true }
 shadow-rs = "0.11.0"
 
 [features]
-default = ["sample_kbc"]
+default = ["sample_kbc", "grpc"]
+grpc = ["tokio", "tonic", "prost", "tonic-build"]
+ttrpc = ["dep:ttrpc", "ttrpc-codegen", "protobuf"]
```

Comment:
Is it OK to change "dep:ttrpc" to "ttrpc"? It's not supported by older rust versions.

Reply:
In the latest version of rust, if we do not add this prefix, the compiler will encounter the following error because the dependency name is the same as the feature name:

```shell
$ cargo build --release --no-default-features --features " sample_kbc ttrpc grpc" --target x86_64-unknown-linux-gnu
error: failed to parse manifest at `/root/data/jiale/ttrpc/attestation-agent/app/Cargo.toml`

Caused by:
  optional dependency `ttrpc` is not included in any feature
  Make sure that `dep:ttrpc` is included in one of features in the [features] table.
make: *** [Makefile:115: build] Error 101
```

Reply:
Maybe we can add a prefix to feature name like `proto_ttrpc` and `proto_grpc` or some better.